scene/gui/popup_menu.cpp

```diff
@@ -566,52 +569,50 @@ void PopupMenu::_input_from_window_internal(const Ref<InputEvent> &p_event) {
 		}
 	}
 
-	Ref<InputEventMouseButton> b = p_event;
+	if (m.is_valid() && drag_to_press) {
+		BitField<MouseButtonMask> initial_button_mask = m->get_button_mask();
+		if (!initial_button_mask.has_flag(mouse_button_to_mask(MouseButton::LEFT)) && !initial_button_mask.has_flag(mouse_button_to_mask(MouseButton::RIGHT))) {
+			mouse_is_pressed = false;
+		}
+
+		if (!item_clickable_area.has_point(m->get_position()) && !mouse_is_pressed) {
+			drag_to_press = false;
+		}
+	}
 
-	if (b.is_valid()) {
-		if (!item_clickable_area.has_point(b->get_position())) {
+	if ((b.is_valid() && b->is_pressed()) || (!mouse_is_pressed && drag_to_press)) {
+		drag_to_press = false;
+
+		// Disable clicks under a time threshold to avoid selection right when opening the popup.
+		uint64_t now = OS::get_singleton()->get_ticks_msec();
+		uint64_t diff = now - popup_time_msec;
+		if (diff < 250) {
```

Comment:
This should take effect only when dragging. If you click twice very fast, the second click is not registered, which makes the menu feel bugged.

Reply:
Really, I'd like to get rid of this entirely since it's typically bad UI design to have the cursor over an item when the menu is expanded, but there are instances where that exist in the editor (like language selection in editor settings, which I personally think should include a scroll container). The reason it was increased from 150 to 250 is because slightly dragging can accidently select items in these menus, and the difference in 100ms is not noticeable in normal circumstances. 

While it exists, I'll make it trigger on dragging. I agree.

Reply:
"Speed Run" users assuming they don't just hotkey everything should be able to drag select things as fast as humanly possible, without the risk of affecting "Click-Click" users, is how I see it, but I think this PR is at least worth testing with the general public until other issues are resolved.